pkg/downloader/downloader.go

```diff
@@ -356,16 +356,63 @@ func (d *GitDownloader) Download(opts DownloadOptions) error {
 		return errors.New("git source is nil")
 	}
 
-	_, err := git.CloneWithOpts(
-		git.WithCommit(gitSource.Commit),
-		git.WithBranch(gitSource.Branch),
-		git.WithTag(gitSource.Tag),
-		git.WithRepoURL(gitSource.Url),
-		git.WithLocalPath(opts.LocalPath),
-	)
+	// If caching is enabled, clone the repo as a bare repository to the cache.
+	if opts.EnableCache {
+		// Check if the bare repository already exists in the cache.
+		if utils.DirExists(opts.CachePath) {
```

Comment:
Hi @Manoramsharma 😄 

 are some logical issues here that need to be considered:

1. Whether the bare repository exists.
2. Whether the corresponding `tag/commit/branch` exists in the bare repository.

Only when both conditions are met, that **the bare repository exists** and **that the corresponding tag/commit/branch exists in the repository** will the cache be hit, and the corresponding content will be cloned from the cache. Your current if statement only considers the first condition. The second condition needs to be added.